anchor-tx/src/cas_remote.rs

```diff
@@ -0,0 +1,325 @@
+use std::time::Duration;
+
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use base64::{
+    engine::general_purpose::{STANDARD_NO_PAD as b64_standard, URL_SAFE_NO_PAD as b64},
+    Engine as _,
+};
+use futures::TryStreamExt;
+use multihash_codetable::{Code, MultihashDigest};
+use ring::signature::Ed25519KeyPair;
+use serde::{Deserialize, Serialize};
+use tokio::time::interval;
+use tracing::{debug, warn};
+use uuid::Uuid;
+
+use ceramic_anchor_service::{
+    DetachedTimeEvent, MerkleNode, MerkleNodes, Receipt, TransactionManager,
+};
+use ceramic_car::CarReader;
+use ceramic_core::{
+    cid_from_ed25519_key_pair, did_key_from_ed25519_key_pair, Cid, StreamId, StreamIdType,
+};
+use ceramic_event::unvalidated::Proof;
+
+pub const AGENT_VERSION: &str = concat!("ceramic-one/", env!("CARGO_PKG_VERSION"));
+
+#[derive(Serialize, Deserialize, Debug)]
+struct CasAuthPayload {
+    url: String,
+    nonce: String,
+    digest: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct CasAnchorRequest {
+    stream_id: StreamId,
+    cid: String,
+    timestamp: String,
+    ceramic_one_version: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Header {
+    pub alg: String,
+    pub kid: String,
+    pub typ: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Claims {
+    nonce: String,
+    url: String,
+    digest: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct AnchorResponse {
+    pub message: String,
+    pub witness_car: Option<String>,
+}
+
+fn cid_to_stream_id(cid: Cid) -> StreamId {
+    StreamId {
+        r#type: StreamIdType::Unloadable,
+        cid,
+    }
+}
+
+/// Remote CAS transaction manager
+pub struct RemoteCas {
+    signing_key: Ed25519KeyPair,
+    url: String,
+    poll_interval: Duration,
+    poll_retry_count: u32,
+    jws_header_b64: String,
+    http_client: reqwest::Client,
+    /// The did:key of the node represented as a StreamId
+    node_stream_id: StreamId,
+}
+
+enum CasResponseParseResult {
+    Anchored(Box<Receipt>),
+    Unauthorized,
+}
+
+#[async_trait]
+impl TransactionManager for RemoteCas {
+    async fn make_proof(&self, root: Cid) -> Result<Receipt> {
```

Comment:
nitpick: I don't love the name "make_proof" for this function.  To me that makes it sound like it's just constructing a proof in memory, rather than that it's actually making a remote call to get the actual work of anchoring to happen.

maybe `anchor_root`?